html/ts/types.ts

```diff
@@ -0,0 +1,75 @@
+// Stuff that Shake generates and injects in
+
+// The version of Shake
+declare const version: string;
+declare const generated: string;
+
+/////////////////////////////////////////////////////////////////////
+// PROFILE DATA
+
+type timestamp = int;
+
+type pindex = int; // an index into the list of profiles
+
+interface Profile {
+    index: pindex; // My index in the list of profiles
+    name: string; // Name of the thing I built
+    execution: seconds; // Seconds I took to execute
+    built: int; // number of times built
+    filesWritten: string[]; // files written
+    filesRead: string[]; // files read
+    readers: pindex[]; // cmds that depend on me (aka they read file(s) i wrote)
+    writers: pindex[]; // cmds I depend on (aka i read file(s) they wrote)
+    hazards: pindex[]; // cmds that violate consistency with this cmd.
+}
+
+function untraced(p: Profile): seconds {
+    return 0;
+}
+
+type ProfileRaw =
+    [ string
+    , seconds
+    , int
+    , string[]
+    , string[]
+    , pindex[]
+    , pindex[]
+    , pindex[]
+    ];
+
+/////////////////////////////////////////////////////////////////////
```

Comment:
This declaration and interface are not necessary

Reply:
I'm sorry I don't understand, can you explain?

Reply:
The whole `progress` feature is not present in Rattle, so `progress` the function and `Progress` the type are no longer required.